include/zephyr/arch/common/pm_s2ram.h

```diff
@@ -76,6 +81,11 @@ void pm_s2ram_mark_set(void);
  *
  * Default implementation is checking a magic word in RAM. CONFIG_PM_S2RAM_CUSTOM_MARKING
  * allows custom implementation.
+ * The following requirements must be fulfilled:
+ * - the function cannot use stack (most likely asm function)
+ * - the content of the R1, R4 and LR registers must remain unchanged
```

Comment:
why R4?

Reply:
> why R4?

The content of R0 was moved there:
https://github.com/zephyrproject-rtos/zephyr/pull/80039/files#diff-011b9cab66d0e92083260695202e01cc69700c2b3283e29ae196735bfc39c753L31-L32
And restored after calling the function:
https://github.com/zephyrproject-rtos/zephyr/pull/80039/files#diff-011b9cab66d0e92083260695202e01cc69700c2b3283e29ae196735bfc39c753R151